ui/tests/app/(index)/_components/AfterLogin.test.tsx

```diff
@@ -0,0 +1,129 @@
+import Role from '@/access/Role'
+import { render, screen } from '@testing-library/react';
+import AfterLogin from '@/app/(index)/_components/AfterLogin';
+import User from '@/access/User';
+
+const testUser: User = JSON.parse(process.env.TEST_USER_A as string);
+describe('AfterLogin', () => {
+    const numberOfGroupings = 18
+    const numberOfMemberships = 17
+
+    const admin: User = {
+        ...testUser,
+        roles:[Role.UH, Role.OWNER,Role.ADMIN] as const
+
+    }
+
+    const owner: User = {
+        ...testUser,
+        roles: [Role.UH, Role.OWNER] as const
+    }
+
+    const uhUser: User = {
+        ...testUser,
+        roles: [Role.UH] as const
+    }
+
+    const getHighestRole = (user: User) => {
+        if (user.roles.includes(Role.ADMIN)) return 'Admin';
+        else if (user.roles.includes(Role.OWNER)) return 'Owner';
+        else return 'Member';
+    }
```

Comment:
Using a function defined in both the test and the actual file is not the best way to test. This makes it fragile, if you edit the code in the actual file, you need to change it down here. It might be preferred to have lay out all the expect codes rather than have helper functions.

Reply:
done this and all above